t_clickhouse/test_clickhouse_logs.py

```diff
@@ -0,0 +1,307 @@
+"""
+Verify tfw_logger logging
+"""
+
+from datetime import datetime, timezone
+from ipaddress import IPv4Address
+
+from helpers import tf_cfg
+from helpers.clickhouse import ClickHouseFinder
+from test_suite import tester
+
+__author__ = "Tempesta Technologies, Inc."
+__copyright__ = "Copyright (C) 2018-2025 Tempesta Technologies, Inc."
+__license__ = "GPL2"
+
+
+class TestClickhouseLogsBaseTest(tester.TempestaTest):
+    backends = [
+        {
+            "id": "deproxy",
+            "type": "deproxy",
+            "port": "8000",
+            "response": "static",
+            "response_content": "HTTP/1.1 201 OK\r\n" "Content-Length: 8\r\n\r\n12345678",
+        },
+    ]
+    tempesta = dict(
+        config="""
+            listen 80;
+            server ${server_ip}:8000;
+
+            access_log dmesg mmap mmap_host=${tfw_logger_clickhouse_host} mmap_log=${tfw_logger_daemon_log};
+        """
+    )
+    clients = [
+        {
+            "id": "deproxy",
+            "type": "deproxy",
+            "addr": "${tempesta_ip}",
+            "port": "80",
+        }
+    ]
+
+    @staticmethod
+    def send_simple_request(deproxy_client, request=None, expected_status: str = "201") -> None:
+        """
+        The simple request with 200-code response
+        """
+        deproxy_client.send_request(
+            request=request or deproxy_client.create_request(method="GET", headers=[]),
+            expected_status_code=expected_status,
+            timeout=10,
+        )
+
+    def setUp(self):
+        super(TestClickhouseLogsBaseTest, self).setUp()
+        self.start_all_services(client=False)
+
+
+class TestClickhouseLogsBufferConfiguration(TestClickhouseLogsBaseTest):
+    tempesta = dict(
+        config="""
+            listen 80;
+            server ${server_ip}:8000;
+
+            mmap_log_buffer_size 4096;
+            access_log dmesg mmap mmap_host=${tfw_logger_clickhouse_host} mmap_log=${tfw_logger_daemon_log};
+        """
+    )
+
+    def test_mmap_buffer(self):
+        """
+        Check the buffer works fine
+        """
+        client = self.get_client("deproxy")
+        client.start()
+
+        self.send_simple_request(client)
+        self.assertWaitUntilEqual(lambda: self.loggers.dmesg.access_log_records_count(), 1)
+
+        client.make_requests([client.create_request(method="GET", headers=[])] * 4100)
+
+        self.assertTrue(client.wait_for_response())
+        # buffer size + simple request before
+        self.assertWaitUntilEqual(lambda: self.loggers.clickhouse.access_log_records_count(), 4097)
```

Comment:
You can just use `self.loggers.clickhouse.access_log_records_count`. Why do use lambda here?
```suggestion
        self.assertWaitUntilEqual(self.loggers.clickhouse.access_log_records_count, 4097)
```